enzyme/Enzyme/MLIR/Interfaces/CloneFunction.cpp

```diff
@@ -27,7 +27,11 @@ getFunctionTypeForClone(mlir::FunctionType FTy, DerivativeMode mode,
   for (auto &&[Ty, returnPrimal, returnShadow, activity] : llvm::zip(
            FTy.getResults(), returnPrimals, returnShadows, ReturnActivity)) {
     if (returnPrimal) {
-      RetTypes.push_back(Ty);
+      if (width != 1) {
+        RetTypes.push_back(mlir::RankedTensorType::get({width}, Ty));
```

Comment:
This shouldn’t need changing since the primal is always unmodified, only Derivatives are changed (and we should be pushing the getshadow types for those below)

Reply:
Oh, then I'm confused of what batched autodiff is.
How should my testcase change?

Reply:
Nvm, it clicked. It's just the shadow that's batched 😅

Reply:
so here's an example from llvm vector mode for example: https://github.com/EnzymeAD/Enzyme/blob/main/enzyme/test/Enzyme/ForwardModeVector/add.ll

Reply:
tho perhaps mul will be more illustrative, https://github.com/EnzymeAD/Enzyme/blob/main/enzyme/test/Enzyme/ForwardModeVector/mul.ll (and obviously feel free to look at any/all of the other examples